File: java/src/main/java/view/TelaAnalise.java
package view;

import model.Time;
import org.jfree.chart.ChartFactory;
import org.jfree.chart.ChartPanel;
import org.jfree.chart.JFreeChart;
import org.jfree.chart.plot.PlotOrientation;
import org.jfree.chart.plot.XYPlot;
import org.jfree.chart.renderer.xy.XYLineAndShapeRenderer;
import org.jfree.data.xy.XYSeries;
import org.jfree.data.xy.XYSeriesCollection;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionListener;
import java.util.List;

public class TelaAnalise extends JFrame {
    private JPanel painelEstatisticas;
    private JPanel painelGrafico;
    private JButton btnVoltar;
    private JLabel lblTitulo;
    
    public TelaAnalise() {
        initComponents();
        setupLayout();
    }
    
    private void initComponents() {
        setTitle("Análise Detalhada do Time");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setSize(1200, 700);
        setLocationRelativeTo(null);
        
        lblTitulo = new JLabel("", SwingConstants.CENTER);
        lblTitulo.setFont(new Font("Arial", Font.BOLD, 20));
        lblTitulo.setForeground(new Color(0, 100, 0));
        
        painelEstatisticas = new JPanel();
        painelEstatisticas.setBackground(Color.WHITE);
        painelEstatisticas.setBorder(BorderFactory.createTitledBorder("Estatísticas do Time"));
        
        painelGrafico = new JPanel(new BorderLayout());
        painelGrafico.setBorder(BorderFactory.createTitledBorder("Evolução na Tabela"));
        
        btnVoltar = new JButton("Voltar");
        btnVoltar.setFont(new Font("Arial", Font.PLAIN, 14));
    }
    
    private void setupLayout() {
        setLayout(new BorderLayout());
        
        // Título no topo
        add(lblTitulo, BorderLayout.NORTH);
        
        // Painel central com estatísticas e gráfico
        JPanel painelCentral = new JPanel(new GridLayout(1, 2, 10, 10));
        painelCentral.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
        
        painelCentral.add(painelEstatisticas);
        painelCentral.add(painelGrafico);
        
        add(painelCentral, BorderLayout.CENTER);
        
        // Botão voltar na parte inferior
        JPanel painelBotao = new JPanel(new FlowLayout());
        painelBotao.add(btnVoltar);
        add(painelBotao, BorderLayout.SOUTH);
    }
    
    public void exibirAnalise(Time time, int ano) {
        lblTitulo.setText(String.format("Análise: %s - Campeonato %d", time.getNome(), ano));
        
        // DEBUG: Imprimir histórico no console
        time.imprimirHistorico();
        
        // Atualizar painel de estatísticas
        atualizarEstatisticas(time);
        
        // Atualizar gráfico
        atualizarGrafico(time);
    }
    
    private void atualizarEstatisticas(Time time) {
        painelEstatisticas.removeAll();
        painelEstatisticas.setLayout(new GridLayout(0, 1, 5, 5));
        painelEstatisticas.setBorder(BorderFactory.createCompoundBorder(
            BorderFactory.createTitledBorder("Estatísticas do Time"),
            BorderFactory.createEmptyBorder(10, 10, 10, 10)
        ));
        
        // Criar labels com as estatísticas
        painelEstatisticas.add(criarLabelEstatistica("Time:", time.getNome()));
        painelEstatisticas.add(criarLabelEstatistica("Jogos:", String.valueOf(time.getJogos())));
        painelEstatisticas.add(criarLabelEstatistica("Pontos:", String.valueOf(time.getPontos())));
        painelEstatisticas.add(criarLabelEstatistica("Vitórias:", String.valueOf(time.getVitorias())));
        painelEstatisticas.add(criarLabelEstatistica("Empates:", String.valueOf(time.getEmpates())));
        painelEstatisticas.add(criarLabelEstatistica("Derrotas:", String.valueOf(time.getDerrotas())));
        painelEstatisticas.add(criarLabelEstatistica("Gols Feitos:", String.valueOf(time.getGolsFeitos())));
        painelEstatisticas.add(criarLabelEstatistica("Gols Sofridos:", String.valueOf(time.getGolsSofridos())));
        painelEstatisticas.add(criarLabelEstatistica("Saldo de Gols:", String.valueOf(time.getSaldoGols())));
        painelEstatisticas.add(criarLabelEstatistica("Aproveitamento:", 
            String.format("%.1f%%", time.getAproveitamento())));
        
        painelEstatisticas.revalidate();
        painelEstatisticas.repaint();
    }
    
    private JLabel criarLabelEstatistica(String titulo, String valor) {
        JLabel label = new JLabel(String.format("<html><b>%s</b> %s</html>", titulo, valor));
        label.setFont(new Font("Arial", Font.PLAIN, 14));
        return label;
    }
    
    private void atualizarGrafico(Time time) {
        painelGrafico.removeAll();
        
        // Criar dataset para o gráfico
        XYSeries series = new XYSeries("Posição na Tabela");
        
        List<Integer> rodadas = time.getHistoricoRodadas();
        List<Integer> posicoes = time.getHistoricoPosicoes();
        
        System.out.println("=== DADOS DO GRÁFICO ===");
        System.out.println("Time: " + time.getNome());
        System.out.println("Rodadas: " + rodadas);
        System.out.println("Posições: " + posicoes);
        
        if (rodadas.size() != posicoes.size()) {
            System.err.println("ERRO: Tamanhos diferentes! Rodadas: " + rodadas.size() + 
                             ", Posições: " + posicoes.size());
        }
        
        // Adicionar pontos ao gráfico
        for (int i = 0; i < Math.min(rodadas.size(), posicoes.size()); i++) {
            series.add(rodadas.get(i), posicoes.get(i));
            System.out.printf("Adicionando ponto: Rodada %d, Posição %d%n", 
                            rodadas.get(i), posicoes.get(i));
        }
        
        if (series.getItemCount() == 0) {
            // Se não há dados, mostrar mensagem
            JLabel lblSemDados = new JLabel("Dados não disponíveis para o gráfico", SwingConstants.CENTER);
            lblSemDados.setFont(new Font("Arial", Font.BOLD, 16));
            lblSemDados.setForeground(Color.RED);
            painelGrafico.add(lblSemDados, BorderLayout.CENTER);
        } else {
            XYSeriesCollection dataset = new XYSeriesCollection(series);
            
            // Criar gráfico
            JFreeChart chart = ChartFactory.createXYLineChart(
                "Evolução da Posição na Tabela",
                "Rodada",
                "Posição",
                dataset,
                PlotOrientation.VERTICAL,
                true,
                true,
                false
            );
            
            // Personalizar gráfico
            XYPlot plot = chart.getXYPlot();
            plot.getRangeAxis().setInverted(true); // Inverter eixo Y (1º lugar no topo)
            plot.setBackgroundPaint(Color.WHITE);
            plot.setRangeGridlinePaint(Color.LIGHT_GRAY);
            plot.setDomainGridlinePaint(Color.LIGHT_GRAY);
            
            // Melhorar aparência da linha
            XYLineAndShapeRenderer renderer = new XYLineAndShapeRenderer();
            renderer.setSeriesShapesVisible(0, true);
            renderer.setSeriesShapesFilled(0, true);
            renderer.setSeriesStroke(0, new BasicStroke(2.0f));
            plot.setRenderer(renderer);
            
            ChartPanel chartPanel = new ChartPanel(chart);
            chartPanel.setPreferredSize(new Dimension(500, 400));
            
            painelGrafico.add(chartPanel, BorderLayout.CENTER);
        }
        
        painelGrafico.revalidate();
        painelGrafico.repaint();
        System.out.println("=== FIM DADOS GRÁFICO ===");
    }
    
    public void setVoltarListener(ActionListener listener) {
        btnVoltar.addActionListener(listener);
    }
}

File: python/view/tela_analise.py
import tkinter as tk
from tkinter import ttk
import matplotlib.pyplot as plt
from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
from matplotlib.figure import Figure
from typing import Callable
from model.time import Time

class TelaAnalise:
    def __init__(self):
        self.window = tk.Toplevel()
        self.window.title("Análise Detalhada do Time")
        self.window.geometry("1200x700")
        self.window.resizable(True, True)
        
        self._init_components()
        self._setup_layout()
        
        # Callback
        self._voltar_callback = None
    
    def _init_components(self):
        """Inicializa os componentes da interface"""
        # Título
        self.lbl_titulo = tk.Label(
            self.window,
            text="",
            font=("Arial", 20, "bold"),
            fg="#006400"
        )
        
        # Frame principal horizontal
        self.frame_principal = tk.Frame(self.window)
        
        # Frame estatísticas (lado esquerdo)
        self.frame_estatisticas = tk.LabelFrame(
            self.frame_principal,
            text="Estatísticas do Time",
            font=("Arial", 12, "bold"),
            padx=10,
            pady=10
        )
        
        # Frame gráfico (lado direito)
        self.frame_grafico = tk.LabelFrame(
            self.frame_principal,
            text="Evolução na Tabela",
            font=("Arial", 12, "bold"),
            padx=10,
            pady=10
        )
        
        # Botão voltar
        self.btn_voltar = tk.Button(
            self.window,
            text="Voltar",
            font=("Arial", 14),
            width=10,
            command=self._on_voltar
        )
    
    def _setup_layout(self):
        """Configura o layout da interface usando pack()"""
        # Título
        self.lbl_titulo.pack(pady=10)
        
        # Frame principal (horizontal)
        self.frame_principal.pack(fill="both", padx=10, pady=5)
        
        # Frames lado a lado
        self.frame_estatisticas.pack(side="left", fill="y", padx=5)
        self.frame_grafico.pack(side="right", fill="both", expand=True, padx=5)
        
        # Botão voltar
        self.btn_voltar.pack(pady=10)
    
    def _on_voltar(self):
        """Callback para voltar"""
        if self._voltar_callback:
            self._voltar_callback()
    
    def exibir_analise(self, time: Time, ano: int):
        """Exibe análise completa do time"""
        self.lbl_titulo.config(text=f"Análise: {time.nome} - Campeonato {ano}")
        
        # Debug
        time.imprimir_historico()
        
        # Atualizar estatísticas
        self._atualizar_estatisticas(time)
        
        # Atualizar gráfico
        self._atualizar_grafico(time)
    
    def _atualizar_estatisticas(self, time: Time):
        """Atualiza painel de estatísticas"""
        # Limpar frame
        for widget in self.frame_estatisticas.winfo_children():
            widget.destroy()
        
        # Criar labels com estatísticas
        estatisticas = [
            ("Time:", time.nome),
            ("Jogos:", str(time.jogos)),
            ("Pontos:", str(time.pontos)),
            ("Vitórias:", str(time.vitorias)),
            ("Empates:", str(time.empates)),
            ("Derrotas:", str(time.derrotas)),
            ("Gols Feitos:", str(time.gols_feitos)),
            ("Gols Sofridos:", str(time.gols_sofridos)),
            ("Saldo de Gols:", str(time.get_saldo_gols())),
            ("Aproveitamento:", f"{time.get_aproveitamento():.1f}%")
        ]
        
        for i, (titulo, valor) in enumerate(estatisticas):
            frame_stat = tk.Frame(self.frame_estatisticas)
            frame_stat.pack(fill="x", pady=2)
            
            lbl_titulo = tk.Label(
                frame_stat,
                text=titulo,
                font=("Arial", 12, "bold"),
                anchor="w"
            )
            lbl_titulo.pack(side="left")
            
            lbl_valor = tk.Label(
                frame_stat,
                text=valor,
                font=("Arial", 12),
                anchor="w"
            )
            lbl_valor.pack(side="left", padx=(10, 0))
    
    def _atualizar_grafico(self, time: Time):
        """Atualiza o gráfico de evolução"""
        # Limpar frame
        for widget in self.frame_grafico.winfo_children():
            widget.destroy()
        
        rodadas = time.get_historico_rodadas()
        posicoes = time.get_historico_posicoes()
        
        print("=== DADOS DO GRÁFICO ===")
        print(f"Time: {time.nome}")
        print(f"Rodadas: {rodadas}")
        print(f"Posições: {posicoes}")
        
        if not rodadas or len(rodadas) != len(posicoes):
            lbl_sem_dados = tk.Label(
                self.frame_grafico,
                text="Dados não disponíveis para o gráfico",
                font=("Arial", 16, "bold"),
                fg="red"
            )
            lbl_sem_dados.pack(expand=True)
            return
        
        # Criar figura matplotlib
        fig = Figure(figsize=(10, 6), dpi=100)
        ax = fig.add_subplot(111)
        
        # Plotar linha apenas onde há dados
        ax.plot(rodadas, posicoes, 'o-', linewidth=2, markersize=6, color='#1f77b4', label=time.nome)
        
        # Configurar eixos para mostrar escala completa
        ax.set_xlim(0, 39)  # Rodadas de 1 a 38
        ax.set_ylim(21, 0)  # Posições de 1 a 20 (invertido)
        
        # Configurar ticks para mostrar todas as rodadas e posições
        ax.set_xticks(range(1, 39))  # Todas as rodadas de 1 a 38
        ax.set_yticks(range(1, 21))  # Todas as posições de 1 a 20
        
        ax.set_xlabel('Rodada', fontsize=12)
        ax.set_ylabel('Posição', fontsize=12)
        ax.set_title('Evolução da Posição na Tabela', fontsize=14, fontweight='bold')
        ax.grid(True, alpha=0.3)
        ax.legend()
        
        # Adicionar linhas de referência para G4, G6, Z4
        ax.axhline(y=4, color='green', linestyle='--', alpha=0.7)
        ax.axhline(y=6, color='orange', linestyle='--', alpha=0.7)
        ax.axhline(y=17, color='red', linestyle='--', alpha=0.7)
        
        # Adicionar canvas ao frame
        canvas = FigureCanvasTkAgg(fig, self.frame_grafico)
        canvas.draw()
        canvas.get_tk_widget().pack(fill="both", expand=True)
        
        print("=== FIM DADOS GRÁFICO ===")
    
    def set_voltar_listener(self, callback: Callable):
        """Define callback para voltar"""
        self._voltar_callback = callback
    
    def show(self):
        """Mostra a janela"""
        self.window.deiconify()
    
    def hide(self):
        """Esconde a janela"""
        self.window.withdraw()
